### PR TITLE
Add registration form blog

### PR DESCRIPTION
**Pull request by mooreds**
Advanced registration forms blog post.

### CODE REVIEW

File: site/_posts/2020-08-11-advanced-registration-form.md [hunk 1]
```diff
@@ -0,0 +1,174 @@
...
+* First name
+* Email
+* Password
+* Phone number
+
+When you're done, it should look like this: 
+
+{% include _image.liquid src="/assets/img/blogs/advanced-registration-forms/form-first-step.png" alt="Adding fields to our first step." class="img-fluid" figure=false %}
+
+Just as you can add many custom form fields, you can also add as many steps as you want. However, please think of the users who are filling out the form and don't add too many steps!
```
Comment (dpolstra) [inline site/_posts/2020-08-11-advanced-registration-form.md:103]: Maybe add a reference: https://www.searchenginewatch.com/2016/11/14/are-you-asking-for-too-much-information-on-web-forms/

Comment (mooreds) [reply site/_posts/2020-08-11-advanced-registration-form.md:103]: Ooh, good idea. I like Nielsen, will put a link to https://www.nngroup.com/articles/web-form-design/ in there.

File: site/_posts/2020-08-11-advanced-registration-form.md [hunk 2]
```diff
@@ -0,0 +1,176 @@
...
+
+You can mix and match these form fields on a new form. If that's all you need, you can avoid creating custom form fields. But if you need anything new, you'll want to create a new form field.
+
+### Custom form fields
+
+The real power of registration form building can be seen when you add custom fields. You can add as many of these as you'd like. 
+
+You can store values in any of the predefined user fields such as `user.fullName`. But you can also use the `data` field on both the `registration` and the `user` objects to store arbitrary data. 
+
+`registration.data` is the right place to store data related to a user's account but specific to an application. As a reminder, [a registration](/docs/v1/tech/core-concepts/registrations) is a link between a user and an application.
```
Comment (voidmain) [inline site/_posts/2020-08-11-advanced-registration-form.md:52]: I'd reword: "`registration.data` is the right place to store data related to a user's account but specific to an application" to "`registration.data` is the right place to store data related to a user's account that is specific to an application"

File: site/_posts/2020-08-11-advanced-registration-form.md [hunk 3]
```diff
@@ -0,0 +1,176 @@
...
+
+Navigate to the "Applications" tab and create a new FusionAuth application. As a reminder, an application is anything a user can sign into: a web application, native app, API, anything. In this case, you are building out the real estate application; pick something descriptive. How about "Real estate search"? 
+
+You must configure a redirect URL; this is where the user is sent when registration succeeds. To set this up, navigate to the "OAuth" tab of your application and enter "https://fusionauth.io" or another public website. For a production application, of course, you'd instead specify where a user should be sent after registration. Each application may have multiple redirect URLs. Configure where to send a user after successful registration by adding a `redirect_uri` to the registration form link. For this tutorial, you'll only add one and that will be automatically added to the registration URL when you are ready to test drive this from the user's perspective.
```
Comment (voidmain) [inline site/_posts/2020-08-11-advanced-registration-form.md:127]: This is actually the hand off for OAuth and the `redirect_uri` must call the token endpoint. While this is a necessary step, this paragraph doesn't tell the reader anything about OAuth. Perhaps we should link out to another blog or a documentation page that describes `redirect_uri` better? Or we should beef this paragraph up a bit to cover OAuth workflows and how registration is actually a nested flow that is part of the Authorization Code Grant.

File: site/_posts/2020-08-11-advanced-registration-form.md [hunk 4]
```diff
@@ -0,0 +1,176 @@
...
+
+On top of setting up the form, you must configure an application to allow users to register themselves. Otherwise, no users will be allowed to create their own accounts. Navigate to the "Registration" tab and enable "Self service registration". Then check the "Advanced" option and select the form you created. Before you click the "Save" button, you'll see a screen like this:
```
Comment (voidmain) [inline site/_posts/2020-08-11-advanced-registration-form.md:129]: I wonder if "In addition to" would flow better than "On top of".

File: site/_posts/2020-08-11-advanced-registration-form.md [hunk 5]
```diff
@@ -0,0 +1,176 @@
...
+
+On top of setting up the form, you must configure an application to allow users to register themselves. Otherwise, no users will be allowed to create their own accounts. Navigate to the "Registration" tab and enable "Self service registration". Then check the "Advanced" option and select the form you created. Before you click the "Save" button, you'll see a screen like this: 
+
+{% include _image.liquid src="/assets/img/blogs/advanced-registration-forms/application-with-custom-form-configured.png" alt="Specifying a custom registration form for our application." class="img-fluid" figure=false %}
+
+Return to the list of applications. Your form is configured, so the next step is to sign up as a user. 
+
+## Sign up
+
+The moment you've been working towards is here! In just a few clicks, you'll see the glory of your custom registration form.
```
Comment (robotdan) [inline site/_posts/2020-08-11-advanced-registration-form.md:137]: We could mention how they can view the form step(s) using the theme preview for styling or what not - prior to actually going through the registration flow for real. 

`Customizations > Themes > Preview` - select the registration template, and this will allow you to select an application and step to view how it will render.

File: site/_posts/2020-08-11-advanced-registration-form.md [hunk 6]
```diff
@@ -0,0 +1,176 @@
...
+
+If you go to the "User data" tab on the account details page, you can see the information you filled out as well:
+
+{% include _image.liquid src="/assets/img/blogs/advanced-registration-forms/displaying-user-data.png" alt="The user data tab of the newly registered user." class="img-fluid" figure=false %}
+
+You now have a custom registration form and a user can sign up successfully. Nice work! This user information is available from the FusionAuth APIs, in the `registration.data` fields. You can retrieve and modify that data using the [APIs](/docs/v1/tech/apis).
+
+Sorry to end on a down note, but there's trouble in paradise. The placeholders for the fields are not very user friendly. For example, `user.firstName` is an understandable hint, but it'd be better to say "Your first name". The hints also disappear as soon as you type anything in them, which means people might forget what they are putting where. 
+
+You can solve all these problems by modifying the theme. Never fear, doing so will be covered in a future blog post.
```
Comment (robotdan) [inline site/_posts/2020-08-11-advanced-registration-form.md:171]: +1 nice tease for the upcoming post.

File: site/_posts/2020-08-11-advanced-registration-form.md (hunk 3 above)
Comment (mooreds) [reply site/_posts/2020-08-11-advanced-registration-form.md:127]: Hmmm. I wasn't able to find any docs about registration and the `redirect_uri`. I'm not sure about the value of talking about the token endpoint in this blogpost is, but I definitely feel like it should go into part of the guide. What am I missing?

File: site/_posts/2020-08-11-advanced-registration-form.md [hunk 7]
```diff
@@ -0,0 +1,176 @@
...
+
+### Custom form fields
+
+The real power of registration form building can be seen when you add custom fields. You can add as many of these as you'd like. 
+
+You can store values in any of the predefined user fields such as `user.fullName`. But you can also use the `data` field on both the `registration` and the `user` objects to store arbitrary data. 
+
+`registration.data` is the right place to store data related to a user's account but specific to an application. As a reminder, [a registration](/docs/v1/tech/core-concepts/registrations) is a link between a user and an application. 
+
+Since this is a real estate app, the minimum price point of the user is germane to this application. Storing it on the registration is the right approach. If you were later to build a mortgage application using FusionAuth as its auth layer, there'd be different fields, such as loan amount, associated with that registration.
```
[the remark this answers is not in the document]
Comment (mooreds) [reply site/_posts/2020-08-11-advanced-registration-form.md:54]: Sure!

File: site/_posts/2020-08-11-advanced-registration-form.md (hunk 2 above)
Comment (mooreds) [reply site/_posts/2020-08-11-advanced-registration-form.md:52]: Sounds good!

**Comment by mooreds** [comment]
I like the idea of making this a part of the guide. I'd have to retake the screenshots, but it'd do double duty. I'll do that.

File: site/_posts/2020-08-11-advanced-registration-form.md (hunk 5 above)
Comment (mooreds) [reply site/_posts/2020-08-11-advanced-registration-form.md:141]: OMG, that's awesome. I didn't know about that.

Comment (mooreds) [reply site/_posts/2020-08-11-advanced-registration-form.md:141]: I think it makes more sense to put that in the next blog post, which makes some theme changes.
